README.md

```diff
@@ -1,38 +1,146 @@
+| UDP                      | :x:                                      | :white_check_mark: | :white_check_mark:              |
+
+Some of the benefits of using Clash Ethernet are:
+- Every protocol is fully configurable in data width. This is
+  something that other libraries, such as
+  [LiteEth](https://github.com/enjoy-digital/liteeth) and
+  [verilog-ethernet](https://github.com/alexforencich/verilog-ethernet)
+  do not support.
+
+- Clash makes it very easy to combine two or more components, see
+  [Clash Protocols](https://github.com/clash-lang/clash-protocols).
+  For example:
+
+  ```haskell
+  verifyChecksum |> depacketizerC const |> verifyLength
+  ```
+  <!-- Source: src/Clash/Cores/Ethernet/IP/IPPacketizers.hs -->
+
+  This verifies the checksum of a packet, then separates the header
+  from the payload and finally verifies the contents of the header
+  (`verifyLength` is not the most descriptive name for what it does).
+  Doing the same in either Verilog or LiteEth would require
+  significantly more work.
+
+- Every component in Clash Ethernet is fully tested with random input
+  data using
+  [Hedgehog](https://github.com/hedgehogqa/haskell-hedgehog) and
+  [Tasty](https://github.com/UnkindPartition/tasty).
+
+
+Something to consider when choosing to use Clash-ethernet is that, for
+now, it only work on an ECP5 with an RGMII chip. However more FPGA's
+and chips are coming soon™.
 
-# Working with this project
-It's required to have the [nix package manager](https://nixos.org/download.html) working.
+# Resource Usage Comparison
+| Component   | Clash-ethernet | LiteEth | Verilog |
+|-------------|:--------------:|:-------:|:-------:|
+| Total LUT4s | 2768           | 3067    |         |
+| Total DFF's | 936            | 1016    |         |
 
-After installing nix you can obtain a dev/build environment with:
+The table above shows resource usage of Clash Ethernet compared to
+LithEth and Verilog, using a data width of 4 and a simple echo design.
+The data shows that Clash Ethernet is about 10% more efficient in
+terms of DFF's (D-Flip-Flops).
 
-```bash
+TODO: add Verilog resource usage.
+
+# Documentation
+Clash Ethernet uses the [Nix package manager](https://nixos.org/) to setup its build environment. To generate a local html website of the Clash Ethernet documentation
+run the following commands:
+
+```sh
+nix-shell
+cabal haddock
+```
+
+After `cabal` is finished it will print the `<file-path>` to
+an html file to stdout. Open the file in your browser and use it as
+the starting point to the Clash Ethernet documentation.
+
+To get inspired, there are some examples for using Clash Ethernet in the
+`examples/` directory.
+
+For more information look at
+[Clash-protocols](https://github.com/clash-lang/clash-protocols). This
+is the overarching framework of which Clash Ethernet is a part.
+
+# How to use as a developer
```

Comment:
I would include an additional paragraph on where (in `env.sh`, that is) to set up environment variables (interface name and device name). This is a step seasoned developers would understand, but which junior developers might want to get reminded about.